fakebook/src/app/component/main-view/main-view.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { AuthService } from '../../service/auth.service';
+
+@Component({
+  selector: 'app-main-view',
+  templateUrl: './main-view.component.html',
+  styleUrls: ['./main-view.component.css']
+})
+export class MainViewComponent implements OnInit {
+  constructor(private oktaAuth: AuthService, private router: Router) {
+  }
+
+  async ngOnInit(): Promise<void> {
+    this.checkAuthentication();
+  }
+
+  checkAuthentication(): void {
+    // Contains all the logic in proj 2 fakebook ngOnInit. Moved this to test this functionality.
```

Comment:
We did actually make this method originally because we were having difficulty testing the ngOnInit and moving the functionality we had into a separate function helped solve our problem, but we also decided that it wasn't necessarily a bad idea to just call the method instead of having the code be in ngOnInit. We believed this separation was good for clarity as an added benefit of helping us unit test.

We could be wrong and the functionality that is going to be put into this method should just be in the ngOnInit, but as far as we know it was not a problem.